File: Team1/RiyaSingh/countAndSay.java
class Solution {
    public String countAndSay(int n) {
        if (n == 1)     return "1"; 
        if (n == 2)     return "11"; 
        String str = "11";  
        for (int i = 3; i <= n; i++) 
        {
            str += '$'; 
            int len = str.length(); 
  
            int cnt = 1;
            String tmp = "";
            char []arr = str.toCharArray(); 
            for (int j = 1; j < len; j++) 
            {
                if (arr[j] != arr[j - 1]) 
                {
                    tmp += cnt + 0;
                    tmp += arr[j - 1]; 
                    cnt = 1; 
                }
                else cnt++; 
            }
            str = tmp; 
        }
        return str; 
    }
}

File: CountAndSay.cpp
class Solution {
public:
   string solve(string s){
        int count=0;
        string output="";
        for(int i=0; i<s.size(); i++){
            count++;
            if(i== s.size()-1 || s[i]!=s[i+1]){
                output+=(char)(count + '0');
                output+=s[i];
                count=0;
            }
        }
        return output;
    }
    
    string countAndSay(int n) {
        if(n==1) return "1";
        return solve(countAndSay(n-1));
    }
};
